mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -803,13 +829,17 @@ private class LogisticAggregator(
     this
   }
 
-  def count: Long = totalCnt
-
-  def loss: Double = lossSum / totalCnt
+  def loss: Double = {
+    require(weightSum >= 1.0, s"The effective number of samples should be " +
```

Comment:
@rotationsymmetry  This definition requires sum of weights == 1. I was originally thinking about to use this definition, but that will not be compatible with unbiased version as @mengxr pointed out. What users want may be weight=2 means this instance appears in the training dataset twice.